query.go

```diff
@@ -0,0 +1,95 @@
+package qdb
+
+/*
+	#include <qdb/query.h>
+	#include <stdlib.h>
+*/
+import "C"
+import (
+	"bytes"
+	"errors"
+	"strconv"
+	"unsafe"
+)
+
+// Query : a building type to execute a query
+// Retrieves all entries’ aliases that match the specified query.
+// For the complete grammar, please refer to the documentation.
+// Queries are transactional.
+// The complexity of this function is dependent on the complexity of the query.
+type Query struct {
+	HandleType
+	tagsExcluded []string
+	tags         []string
+	types        []string
+}
+
+// Tag : Adds a tag to include into the current query results
+func (q *Query) Tag(t string) *Query {
+	q.tags = append(q.tags, t)
+	return q
+}
+
+// NotTag : Adds a tag to exclude from the current query results
+func (q *Query) NotTag(t string) *Query {
+	q.tagsExcluded = append(q.tagsExcluded, t)
+	return q
+}
+
+// Type : Restrict the query results to a particular type
+func (q *Query) Type(t string) *Query {
+	q.types = append(q.types, t)
+	return q
+}
+
+func (q Query) buildQuery() (string, error) {
+	var query bytes.Buffer
+	for idx, t := range q.tags {
```

Comment:
`strings.join(q.tags, " and ")` ?

Reply:
strings.Join(func(t string){ "tag=" + strconv.Quote(q.tags) }(tag), " and ")